tests/Gedmo/Timestampable/Fixture/Attribute/TitledArticle.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+namespace Gedmo\Tests\Timestampable\Fixture\Attribute;
+
+use Doctrine\DBAL\Types\Types;
+use Doctrine\ORM\Mapping as ORM;
+use Gedmo\Mapping\Annotation as Gedmo;
+use Gedmo\Timestampable\Timestampable;
+
+#[ORM\Entity]
+class TitledArticle implements Timestampable
+{
+    #[ORM\Id]
+    #[ORM\GeneratedValue]
+    #[ORM\Column(type: Types::INTEGER)]
+    private $id;
+
+    #[ORM\Column(name: 'title', type: Types::STRING, length: 128)]
+    private $title;
+
+    #[ORM\Column(name: 'text', type: Types::STRING, length: 128)]
+    private $text;
+
+    #[ORM\Column(name: 'state', type: Types::STRING, length: 128)]
+    private $state;
+
+    /**
+     * @var \DateTime
+     */
+    #[ORM\Column(name: 'chtext', type: Types::DATETIME_MUTABLE, nullable: true)]
+    #[Gedmo\Timestampable(on: 'change', field: 'text')]
+    private $chText;
+
+    /**
+     * @var \DateTime
+     */
+    #[ORM\Column(name: 'chtitle', type: Types::DATETIME_MUTABLE, nullable: true)]
+    #[Gedmo\Timestampable(on: 'change', field: 'title')]
+    private $chTitle;
+
+    /**
+     * @var \DateTime
+     */
+    #[ORM\Column(name: 'closed', type: TYPES::DATETIME_MUTABLE, nullable: true)]
+    #[Gedmo\Timestampable(on: 'change', field: 'state', value: ['Published', 'Closed'])]
+    private $closed;
+
+    /**
+     * @param \DateTime $chtext
```

Comment:
Probably, I just tried to stick to the existing standard. Usually I run into wild behaviors when fields aren't declared as nullable, even if their columns are mapped as non-nullable, though, so it might get somewhat confusing.

Reply:
I understand. We are trying to enforce the typing in our current codebase, so (ideally), all the new code should follow these rules.